src/lsptoolshost/restore.ts

```diff
@@ -22,10 +28,15 @@ export function registerRestoreCommands(
     );
     context.subscriptions.push(
         vscode.commands.registerCommand('dotnet.restore.all', async (): Promise<void> => {
-            return restore(languageServer, restoreChannel);
+            return restore(languageServer, restoreChannel, [], true);
         })
     );
+
+    languageServer.registerOnRequest(UnresolvedProjectDependenciesNotification.type, async (params) => {
```

Comment:
Is this a request or a notification? Or I guess the underlying type is a void which means notification?

Reply:
Its a request that returns void (but waits until completion) - there are separate APIs for notifications